core/src/main/java/org/apache/iceberg/ManifestReader.java

```diff
@@ -282,14 +283,18 @@ private static boolean requireStatsProjection(Expression rowFilter, Collection<S
 
   static boolean dropStats(Expression rowFilter, Collection<String> columns) {
     // Make sure we only drop all stats if we had projected all stats
-    // We do not drop stats even if we had partially added some stats columns
-    return rowFilter != Expressions.alwaysTrue() &&
-        columns != null &&
-        !columns.containsAll(ManifestReader.ALL_COLUMNS) &&
-        Sets.intersection(Sets.newHashSet(columns), STATS_COLUMNS).isEmpty();
+    // We do not drop stats even if we had partially added some stats columns, except for record_count column.
+    // Since we don't want to keep stats map which could be huge in size just because we select record_count, which
+    // is a primitive type.
+    if (rowFilter != Expressions.alwaysTrue() && columns != null &&
+        !columns.containsAll(ManifestReader.ALL_COLUMNS)) {
+      Set<String> interaction = Sets.intersection(Sets.newHashSet(columns), STATS_COLUMNS);
```

Comment:
Did you mean `interaction` or `intersection`?

Reply:
Yes, thanks for catching it!